File: 1-TwoSum/1-TwoSum.py
# Last updated: 8/2/2025, 4:54:25 PM
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        numsDict = {}
        for i in range(len(nums)):
            numsDict[nums[i]] = i
        for i in range(len(nums)):
            if (target - nums[i]) in numsDict and not i == numsDict[target - nums[i]]:
                return [i, numsDict[target - nums[i]]]
        

File: 1-TwoSum/1-TwoSum.java
// Last updated: 8/2/2025, 4:54:24 PM
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<>();
        int[] res = new int[2];
        int n = nums.length;
        for(int i = 0; i < n; i++) {
            if(map.containsKey(target - nums[i])) {
                res[0] = i;
                res[1] = map.get(target - nums[i]);
                break;
            } else {
                map.put(nums[i], i);
            }
        }
        return res;
    }
}